cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -110,36 +102,31 @@ public class AppFabricServer extends AbstractIdleService {
   private CommonNettyHttpServiceFactory commonNettyHttpServiceFactory;
 
   /**
-   * Construct the AppFabricServer with service factory and cConf coming from guice injection.
+   * Construct the AppFabricServer with service factory and cConf coming from guice
+   * injection.
    */
   @Inject
   public AppFabricServer(CConfiguration cConf, SConfiguration sConf,
-                         DiscoveryService discoveryService,
-                         @Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname,
-                         @Named(Constants.AppFabric.HANDLERS_BINDING) Set<HttpHandler> handlers,
-                         @Nullable MetricsCollectionService metricsCollectionService,
-                         ProgramRuntimeService programRuntimeService,
-                         RunRecordCorrectorService runRecordCorrectorService,
-                         ProgramRunStatusMonitorService programRunStatusMonitorService,
-                         ApplicationLifecycleService applicationLifecycleService,
-                         ProgramNotificationSubscriberService programNotificationSubscriberService,
-                         ProgramStopSubscriberService programStopSubscriberService,
-                         @Named("appfabric.services.names") Set<String> servicesNames,
-                         @Named("appfabric.handler.hooks") Set<String> handlerHookNames,
-                         AuditLogSubscriberService auditLogSubscriberService,
-                         CoreSchedulerService coreSchedulerService,
-                         CredentialProviderService credentialProviderService,
-                         NamespaceCredentialProviderService namespaceCredentialProviderService,
-                         ProvisioningService provisioningService,
-                         BootstrapService bootstrapService,
-                         SystemAppManagementService systemAppManagementService,
-                         TransactionRunner transactionRunner,
-                         RunRecordMonitorService runRecordCounterService,
-                         CommonNettyHttpServiceFactory commonNettyHttpServiceFactory,
-                         RunDataTimeToLiveService runDataTimeToLiveService,
-                         SourceControlOperationRunner sourceControlOperationRunner,
-                         RepositoryCleanupService repositoryCleanupService,
-                         OperationNotificationSubscriberService operationNotificationSubscriberService) {
+      DiscoveryService discoveryService,
+      @Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname,
+      @Named(Constants.AppFabric.SERVER_HANDLERS_BINDING) Set<HttpHandler> handlers,
+      @Nullable MetricsCollectionService metricsCollectionService,
+      ProgramRuntimeService programRuntimeService,
+      ProgramRunStatusMonitorService programRunStatusMonitorService,
```

Comment:
Do we want this one to run in multiple copies? What happens if two start running at the same time?

Reply:
Removed from Appfabric Server.

Only one copy can run in Appfabric Processor.